src/collectors/jolokia/kafka_jolokia.py

```diff
@@ -0,0 +1,592 @@
+            m_type = ''
+            name = ''
+            request = ''
+            topic = ''
+            partition = ''
+            broker=''
+            delayed_op=''
+            network_processor=''
+            processor=''
+            client=''
+            broker_host=''
+            broker_port=''
+            group=''
+            thread=''
+            fetcher=''
+
+            # Loop through the array of name-value pairs.
+            for i in range(len(kvps)):
+                # Split the pair on the equal sign such that kvp[0] is the key and kvp[1] is the value
+                kvp = string.split(kvps[i], '=')
+                
+                # Check the name of the key, and set the appropriate variable accordingly.
+                # In most cases, we'll replace any dots in the values with dashes. We don't do this
+                # for the "type" key, though, as in some cases we need to do further parsing first.
+                if (kvp[0].lower() == 'type'):
+                        m_type = kvp[1]
+                elif (kvp[0].lower() == 'name'):
+                        name = '.' + string.replace(kvp[1], '.', '-')
+                elif (kvp[0].lower() == 'request'):
+                        request = '.' + string.replace(kvp[1], '.', '-')
+                elif (kvp[0].lower() == 'topic'):
+                        topic = '.' + string.replace(kvp[1], '.', '-')
+                elif (kvp[0].lower() == 'partition'):
+                        partition = '.' + 'partition-' + string.replace(kvp[1], '.', '-')
+                elif (kvp[0].lower() == 'broker-id'):
+                        broker = '.' + 'broker-' + string.replace(kvp[1], '.', '-')
+                elif (kvp[0].lower() == 'networkprocessor'):
+                        network_processor = '.' + 'networkprocessor-' + string.replace(kvp[1], '.', '-')
+                elif (kvp[0].lower() == 'processor'):
+                        processor = '.' + 'processor-' + string.replace(kvp[1], '.', '-')
+                elif (kvp[0].lower() == 'clientid' or kvp[0].lower() == 'client-id'):
+                        client = '.' + string.replace(kvp[1], '.', '-')
+                elif (kvp[0].lower() == 'delayedoperation'):
+                        delayed_op = '.' + string.replace(kvp[1], '.', '-')
+                elif (kvp[0].lower() == 'brokerhost'):
+                        broker_host = '.' + string.replace(kvp[1], '.', '-')
+                elif (kvp[0].lower() == 'brokerport'):
+                        broker_port = string.replace(kvp[1], '.', '-')
+                elif (kvp[0].lower() == 'groupid'):
+                        group = '.' + string.replace(kvp[1], '.', '-')
+                elif (kvp[0].lower() == 'threadid'):
+                        broker = '.' + 'thread-' + string.replace(kvp[1], '.', '-')
+                elif (kvp[0].lower() == 'fetchertype'):
+                        fetcher = '.' + string.replace(kvp[1], '.', '-')
```

Comment:
No - and it should be. Looks like I missed adding that component into the metric name.  Good catch.  I'll go back and do a another round of testing to validate where that should be placed.